src/esri-loader.ts

```diff
@@ -10,172 +10,19 @@
-          resolve(script);
-        } else {
-          // wait for the script to load and then resolve
-          handleScriptLoad(script, resolve, reject);
-        }
-      }
-    } else {
-      if (isLoaded()) {
-        // the API has been loaded by some other means
-        // potentially trying to load a different version of the API
-        reject(new Error(`The ArcGIS API for JavaScript is already loaded.`));
-      } else {
-        // this is the first time attempting to load the API
-        if (options.css) {
-          // load the css before loading the script
-          loadCss(options.css, options.insertCssBefore);
-        }
-        if (options.dojoConfig) {
-          // set dojo configuration parameters before loading the script
-          window['dojoConfig'] = options.dojoConfig;
-        }
-        // create a script object whose source points to the API
-        script = createScript(url);
-        _currentUrl = url;
-        // once the script is loaded...
-        handleScriptLoad(script, () => {
-          // update the status of the script
-          script.setAttribute('data-esri-loader', 'loaded');
-          // return the script
-          resolve(script);
-        }, reject);
-        // load the script
-        document.body.appendChild(script);
-      }
-    }
-  });
-}
 
-// wrap dojo's require() in a promise
-function requireModules(modules: string[]): Promise<any[]> {
-  return new utils.Promise((resolve, reject) => {
-    // If something goes wrong loading the esri/dojo scripts, reject with the error.
-    const errorHandler = window['require'].on('error', reject);
-    window['require'](modules, (...args) => {
-      // remove error handler
-      errorHandler.remove();
-      // Resolve with the parameters from dojo require as an array.
-      resolve(args);
-    });
-  });
-}
-
-// returns a promise that resolves with an array of the required modules
-// also will attempt to lazy load the ArcGIS API if it has not already been loaded
-export function loadModules(modules: string[], loadScriptOptions: ILoadScriptOptions = {}): Promise<any[]> {
-  if (!isLoaded()) {
-    // script is not yet loaded
-    if (!loadScriptOptions.url && _currentUrl) {
-      // alredy in the process of loading, so default to the same url
-      loadScriptOptions.url = _currentUrl;
-    }
-    // attept to load the script then load the modules
-    return loadScript(loadScriptOptions).then(() => requireModules(modules));
-  } else {
-    // script is already loaded, just load the modules
-    return requireModules(modules);
-  }
-}
-
-export { loadCss } from './utils/css';
+// re-export the functions that are part of the public API
+import { loadModules } from './modules';
+import { getScript, isLoaded, loadScript } from './script';
+import { loadCss } from './utils/css';
+import utils from './utils/index';
+export { getScript, isLoaded, loadModules, loadScript, loadCss, utils };
 
```

Comment:
why not just?

```ts
export { loadModules } from './modules';
export { getScript, isLoaded, loadScript } from './script';
export { loadCss } from './utils/css';
export { utils } from './utils/index';
```

Reply:
b/c they're needed below for the stupid `export default`, so I will switch to `export from` once that is removed in the next breaking change.